pilot/pkg/xds/filters/filters.go

```diff
@@ -164,7 +164,11 @@ var (
 		},
 	}
 
-	tcpMx = protoconv.MessageToAny(&metadata_exchange.MetadataExchange{Protocol: "istio-peer-exchange"})
+	tcpMx = protoconv.TypedStructWithFields("type.googleapis.com/envoy.tcp.metadataexchange.config.MetadataExchange",
```

Comment:
can you add a comment on this as it is from istio extensions, which is lack of doc from envoy

Reply:
Done